### PR TITLE
I2676

### PR DESCRIPTION
**Pull request by weshinsley**
A small PR that allows vault to be called "vault.exe" on windows, both in the extraction from the ZIP file in vault_test_server_install, and in the starting of the server in vault_server_manager_bin.

### CODE REVIEW

File: R/server_manager.R [hunk 1]
```diff
@@ -291,7 +298,12 @@ vault_install <- function(dest, quiet, version) {
     tmp <- tempfile()
     dir_create(tmp)
     utils::unzip(zip, exdir = tmp)
-    ok <- file.copy(file.path(tmp, "vault"), dest_bin)
+
+    ok <- file.copy(file.path(tmp, vault_exe_filename()), dest_bin)
+    if (!ok) {
+      warning(sprintf("Failed to copy vault to %s", dest_bin))
```
Comment (richfitz) [inline R/server_manager.R]: If you want to check a successful copy, please see `file_copy` in orderly (copy it and the tests here)

File: R/server_manager.R [hunk 2]
```diff
@@ -40,6 +40,13 @@ vault_test_server <- function(https = FALSE, init = TRUE,
   vault_server_manager()$new_server(https, init, if_disabled)
 }
 
+vault_exe_filename <- function() {
```
Comment (richfitz) [inline R/server_manager.R]: please rewrite to take an argument `windows = is_windows()` argument, then test both paths in your tests

File: R/server_manager.R (hunk 1 above)
Comment (weshinsley) [reply R/server_manager.R]: Done - added file_copy to R/util.R, with tests in tests/testthat/test-util.R, and modified the above to call it. (The above is the only place where file.copy was called)

File: R/server_manager.R (hunk 2 above)
Comment (weshinsley) [reply R/server_manager.R]: Done - and I switched to using the existing vault_platform() as the source of truth about what platform we're on.

**Pull request merged**